AutoRest/Generators/Extensions/Azure.Extensions/AzureExtensions.cs

```diff
@@ -114,9 +114,7 @@ public static void UpdateHeadMethods(ServiceClient serviceClient)
                 }
                 else
                 {
-                    throw new NotSupportedException(
-                        string.Format(CultureInfo.InvariantCulture, 
-                        Resources.HeadMethodInvalidResponses, method.Name));
+                    Logger.LogWarning(string.Format(CultureInfo.InvariantCulture, Resources.HeadMethodPossibleIncorrectSpecification, method.Name));
```

Comment:
@stankovski @yugangw-msft This is the "fix" I made for Azure header generation.  Let me know what you think.  Most of the rest of the PR is new tests or regenerated test code (and a few bug fixes).